### PR TITLE
basic: lower check for sum to be checked at the end of the loop

### PR DESCRIPTION
**Pull request by SiliconA-Z**
The sum being checked at the end, especially after no match has been found, makes more sense than to check at the top of the loop, which means it is checked even when the sum is 0.

### CODE REVIEW

File: src/basic/process-util.c [hunk 1]
```diff
@@ -1083,6 +1078,10 @@ int getenv_for_pid(pid_t pid, const char *field, char **ret) {
                         *ret = value;
                         return 1;
                 }
+
+                sum += r;
```
Comment (YHNdnzj) [inline src/basic/process-util.c:1082]: Please don't move this. Having this close to where `r` is assigned makes the code easier to follow.

Comment (AZero13) [reply src/basic/process-util.c:1082]: I thought it being closer to sum would make it easier to follow.

Comment (vcaputo) [reply src/basic/process-util.c:1082]: Please stop.

**Review by yuwata — CHANGES_REQUESTED** [review]
Sorry, NAK. The check should be at the beginning. To make the condition always checked even when we will insert `continue` in the loop.
The cost for the check is extremely small. We should not optimize such.

**Pull request closed without merging**